aws/rust-runtime/aws-config/src/lib.rs

```diff
@@ -583,10 +564,13 @@ mod loader {
             let conf = self
                 .provider_config
                 .unwrap_or_else(|| {
-                    ProviderConfig::init(time_source.clone(), sleep_impl.clone())
+                    let mut config = ProviderConfig::init(time_source.clone(), sleep_impl.clone())
                         .with_fs(self.fs.unwrap_or_default())
-                        .with_env(self.env.unwrap_or_default())
-                        .with_http_connector(http_connector.clone())
+                        .with_env(self.env.unwrap_or_default());
+                    if let Some(http_client) = self.http_client.clone() {
+                        config = config.with_http_client(http_client);
+                    }
```

Comment:
I wonder if we should try to encapsulate this pattern somehow...`config.override_client(http_client)` that will take the value if it's some but ignore it if it's `None`

Reply:
It seems like something that would benefit the aws-config code or generated code a lot, but not necessarily customer code. It would also double the number of methods on builders, making docs harder to navigate. What we could do is add a separate trait that has the `override_*` methods on it that you can import.